src/settingsui.ts

```diff
@@ -126,6 +131,51 @@ function saveSettings() {
 	writeFileSync(userPrefsPath, JSON.stringify(userPrefs, null, 2), { encoding: 'utf-8' });
 	ipcRenderer.send('settingsUI_updates_userPrefs', userPrefs); // send them back to main
 }
+function loadUserScriptSettings(eventSuffix: string, settings: { [key: string]: UserscriptRenderReadySetting }): UserPrefs {
+	try {
+		var settingsJSON = JSON.parse(readFileSync(join(userscriptPrefsPath, `${eventSuffix}.json`), 'utf-8'));
+		Object.keys(settingsJSON).forEach(settingKey => {
+			if (// Comparison chain basically validates the settings JSON
+				settingKey in settings && // Make sure setting key is a changeable key
+				typeof settings[settingKey].changed === "function" &&  // Make sure setting key has a changed function
+				settingsJSON[settingKey] !== settings[settingKey].value && // Make sure you aren't applying changes unnecissarily
+				typeof settingsJSON[settingKey] === typeof settings[settingKey].value // Ensure the saved and scripted values are of the same type
+			) {
+				userscriptPreferences[eventSuffix][settingKey] = settingsJSON[settingKey]
+			}
+		})
+		return userscriptPreferences[eventSuffix]
+	} catch (e) { // Settings file for script probably doesn't exist yet
+		return {}
+	}
+}
+function userscriptSettingsResetDefaults(eventSuffix: string, userscriptSettings: { [key: string]: UserscriptRenderReadySetting }, userscriptCategoryID: string, brokenSettings: string[]) {
+	var optionsContainer = document.querySelector(`.${userscriptCategoryID}`)
+	Object.keys(userscriptSettings).forEach(settingKey => {
+		if (!brokenSettings.includes(settingKey)) {
+			var setting = userscriptSettings[settingKey]
+			if (setting.value !== userscriptPreferences[eventSuffix][settingKey]) {
+				setting.changed({ ...setting }.value)
+			}
+			userscriptPreferences[eventSuffix][settingKey] = { ...setting }.value
+			switch (setting.type) {
+				case "bool":
+					//@ts-ignore Typescript doesn't like that I'm not explicitly stating that I'm looking for an input I think... this works perfectly though.
+					optionsContainer.querySelector(`#settingElem-${settingKey}`).querySelector(`.s-update`).checked = { ...setting }.value
```

Comment:
cast it to an input like so / something like this: (i didn't test the suggestion tho)
```suggestion
					const updateElem = (optionsContainer.querySelector(`#settingElem-${settingKey}`).querySelector(`.s-update`) as HTMLInputElement)
					updateElem .checked = { ...setting }.value
```

Reply:
you can also do querySelector<whatever_element_type>

Reply:
yes. https://stackoverflow.com/a/35362350/13342359